openshift-kube-apiserver/admission/autoscaling/managementcpusoverride/admission.go

```diff
@@ -0,0 +1,559 @@
+	plugins.Register(PluginName,
+		func(config io.Reader) (admission.Interface, error) {
+			return &managementCPUsOverride{
+				Handler: admission.NewHandler(admission.Create),
+			}, nil
+		})
+}
+
+// managementCPUsOverride presents admission plugin that should replace pod container CPU requests with a new management resource.
+// It applies to all pods that:
+// 1. are in an allowed namespace
+// 2. and have the workload annotation.
+//
+// It also sets the new management resource request and limit and  set resource annotation that CRI-O can
+// recognize and apply the relevant changes.
+// For more information, see - https://github.com/openshift/enhancements/pull/703
+//
+// Conditions for CPUs requests deletion:
+// 1. The namespace should have allowed annotation "workload.openshift.io/allowed": "management"
+// 2. The pod should have management annotation: "workload.openshift.io/management": "{"effect": "PreferredDuringScheduling"}"
+// 3. All nodes under the cluster should have new management resource - "management.workload.openshift.io/cores"
+// 4. The CPU request deletion will not change the pod QoS class
+type managementCPUsOverride struct {
+	*admission.Handler
+	client         kubernetes.Interface
+	nsLister       corev1listers.NamespaceLister
+	nsListerSynced func() bool
+	nodeLister     corev1listers.NodeLister
+	nodeListSynced func() bool
+}
+
+func (a *managementCPUsOverride) SetExternalKubeInformerFactory(kubeInformers informers.SharedInformerFactory) {
+	a.nsLister = kubeInformers.Core().V1().Namespaces().Lister()
+	a.nsListerSynced = kubeInformers.Core().V1().Namespaces().Informer().HasSynced
+	a.nodeLister = kubeInformers.Core().V1().Nodes().Lister()
+	a.nodeListSynced = kubeInformers.Core().V1().Nodes().Informer().HasSynced
+}
+
+// SetExternalKubeClientSet implements the WantsExternalKubeClientSet interface.
+func (a *managementCPUsOverride) SetExternalKubeClientSet(client kubernetes.Interface) {
+	a.client = client
+}
+
+func (a *managementCPUsOverride) ValidateInitialization() error {
+	if a.client == nil {
+		return fmt.Errorf("%s plugin needs a kubernetes client", PluginName)
+	}
+	if a.nsLister == nil {
+		return fmt.Errorf("%s did not get a namespace lister", PluginName)
+	}
+	if a.nsListerSynced == nil {
+		return fmt.Errorf("%s plugin needs a namespace lister synced", PluginName)
+	}
+	if a.nodeLister == nil {
+		return fmt.Errorf("%s did not get a node lister", PluginName)
+	}
+	if a.nodeListSynced == nil {
+		return fmt.Errorf("%s plugin needs a node lister synced", PluginName)
+	}
+	return nil
+}
+
+func (a *managementCPUsOverride) Admit(ctx context.Context, attr admission.Attributes, o admission.ObjectInterfaces) error {
+	if attr.GetResource().GroupResource() != coreapi.Resource("pods") || attr.GetSubresource() != "" {
+		return nil
+	}
+
+	pod, ok := attr.GetObject().(*coreapi.Pod)
+	if !ok {
+		return admission.NewForbidden(attr, fmt.Errorf("unexpected object: %#v", attr.GetObject()))
+	}
+
```

Comment:
Since static pods reflect the reality of what is running regardless of what admission does to the pod itself, please add a check for static pods here that always returns nil.

Reply:
The problem that mutating the static pod under the kubelet not reflected under the mirror pod, so it consumes the original CPU request under the scheduler. It the main reason why I do not skip mutating of mirror pods.

Reply:
> The problem that mutating the static pod under the kubelet not reflected under the mirror pod, so it consumes the original CPU request under the scheduler. It the main reason why I do not skip mutating of mirror pods.

What you want to do is to update the static pod operators to apply a static pod that contains the correct values.  Having static pods mutated or rejected impedes our ability to know what's actually running.

I believe this will require an external signal, which neatly fixes your other implementation issue.

Reply:
After additional discussion, we will skip the mutation and validation of static pods.